Lib/unittest/test/testmock/testmock.py

```diff
@@ -1702,6 +1702,15 @@ def test_mock_open_dunder_iter_issue(self):
         self.assertEqual(lines[1], 'Norwegian Blue')
         self.assertEqual(list(f1), [])
 
+    def test_mock_open_using_next(self):
+        mocked_open = mock.mock_open(read_data='1st line\n2nd line\n3rd line')
+        f1 = mocked_open('a-name')
+        self.assertEqual(next(f1), '1st line\n')
```

Comment:
Personally, grouping the expected data with variables from assert calls reads nice.  

```python
line1 = next(f1)
line2 = f1.__next__()
# line 3

self.assertEqual(line1, '1st line\n')
# assert for line 2
# assert for line 3
```

Also add a test for `StopIteration` after exhausting f1. You can use

```python
with self.assertRaises(StopIteration):
    next(f1)
```

Reply:
Would prefer moving it to `TestMockOpen` for these tests but would wait for other's opinion on this

Reply:
I've applied the changed you've suggested and also added the missing StopIteration test.

Reply:
Also, would it be more consistent to test `__next__()` prior to `next()` ?

Reply:
Sorry I implied moving the testmock test to testwith. It seems like two copies.

Reply:
Yeah, I've understood that :) but since I see some other already-redundant tests in both files and and since @mariocj89 said testmock and you said testwith.py, I've added both. No offense but I dont know either of you two so I honestly dont who is right. :)

Reply:
Well, it was just an opinion of mine that to have tests grouped under the class. It's up to the core dev who will be merging this.